app/models/message.rb

```diff
@@ -168,7 +169,8 @@ def update_contact_activity
   def dispatch_create_events
     Rails.configuration.dispatcher.dispatch(MESSAGE_CREATED, Time.zone.now, message: self, performed_by: Current.executed_by)
 
-    if outgoing? && conversation.messages.outgoing.count == 1
+    campaign_outgoing_messages = conversation.messages.outgoing.where(created_by: 'Campaign')
+    if outgoing? && (conversation.messages.outgoing.count - campaign_outgoing_messages.count) == 1
```

Comment:
Can't we just query messages where type = outgoing and campaign_id is nil?

Reply:
Updated to : 

outgoing? && conversation.messages.outgoing.where("(additional_attributes->'campaign_id') is null").count == 1